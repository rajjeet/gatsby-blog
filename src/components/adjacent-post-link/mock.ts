export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
