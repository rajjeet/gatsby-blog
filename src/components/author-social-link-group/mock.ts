import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

