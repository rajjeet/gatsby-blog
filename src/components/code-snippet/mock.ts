import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
