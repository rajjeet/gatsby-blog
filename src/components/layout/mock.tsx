import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

