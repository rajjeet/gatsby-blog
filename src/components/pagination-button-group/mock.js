export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
