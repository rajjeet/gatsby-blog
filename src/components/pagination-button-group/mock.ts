export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
