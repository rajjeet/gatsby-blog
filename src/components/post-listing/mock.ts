import { createMockPosts } from '../../utils/testing';
import { TPostEdge } from '../../templates/category-page/types';

export const makeProps = ({
  className = '',
  posts = createMockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  className,
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

export type TProps = {
  className: string;
  posts: TPostEdge[];
  heading?: string;
  numOfPages: number;
  currentPage: number;
  paginationSlug: string;
}
