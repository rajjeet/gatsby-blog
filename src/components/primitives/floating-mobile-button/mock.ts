import { faTimes } from '@fortawesome/free-solid-svg-icons';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  icon: faTimes,
});
