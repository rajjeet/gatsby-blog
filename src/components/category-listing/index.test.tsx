import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { CategoryListing } from './index';
import { createMockGroups } from '../../utils/testing';
import { makeProps } from './mock';

afterEach(cleanup);

describe('<CategoryListing />', () => {
  it('should render', () => {
    const { asFragment } = render(<CategoryListing {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a heading labelled Categories', () => {
    const { getByText } = render(<CategoryListing {...makeProps()} />);
    expect(getByText('Categories')).toBeDefined();
  });

  it('should have two categories with the role of link', () => {
    const { queryAllByRole } = render(<CategoryListing {...makeProps()} />);
    expect(queryAllByRole('link')).toHaveLength(2);
  });

  it('should show categories by name and count', () => {
    const { getByText } = render(<CategoryListing {...makeProps()} />);
    const groups = createMockGroups.categoryGrouping.group;
    groups.forEach((group) => {
      expect(getByText(group.fieldValue)).toBeDefined();
      expect(getByText(group.totalCount.toString())).toBeDefined();
    });
  });
});
