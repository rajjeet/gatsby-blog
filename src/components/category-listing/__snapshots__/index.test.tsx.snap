// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryListing /> should render 1`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH fbiJJR"
  >
    <h5>
      Categories
    </h5>
    <div
      class="sc-bdVaJa fdxsEr"
    >
      <span
        role="link"
        tabindex="0"
      >
        Experience
        <span>
          1
        </span>
      </span>
      <span
        role="link"
        tabindex="0"
      >
        Guide
        <span>
          2
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
